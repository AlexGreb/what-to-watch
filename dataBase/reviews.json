[
  {
    "movieId": 1,
    "reviews": [{"id":1,"user":{"id":17,"name":"Emely"},"rating":6.5,"comment":"This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.","date":"2024-01-07T07:19:48.519Z"}]
  },
  {
    "movieId": 2,
    "reviews":[{"id":1,"user":{"id":12,"name":"Isaac"},"rating":6.9,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-14T07:19:48.519Z"}]
  },
  {
    "movieId": 3,
    "reviews":[{"id":1,"user":{"id":14,"name":"Corey"},"rating":9,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-08T07:19:48.519Z"}]
  },
  {
    "movieId": 4,
    "reviews":[]
  },
  {
    "movieId": 5,
    "reviews":[{"id":1,"user":{"id":16,"name":"Mollie"},"rating":5.7,"comment":"This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.","date":"2024-01-19T07:19:48.519Z"}]
  },
  {
    "movieId": 6,
    "reviews":[]
  },
  {
    "movieId": 7,
    "reviews":[{"id":1,"user":{"id":10,"name":"Max"},"rating":1.2,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-07T07:19:48.519Z"}]
  },
  {
    "movieId": 8,
    "reviews":[]
  },
  {
    "movieId": 9,
    "reviews":[]
  },
  {
    "movieId": 10,
    "reviews":[]
  },
  {
    "movieId": 11,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":5.4,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-22T07:19:48.519Z"}]
  },
  {
    "movieId": 12,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":3.8,"comment":"I really find it difficult to believe this movie is rated highly by people. It's hands down the worst movie I've seen in my life","date":"2024-01-24T07:19:48.519Z"}]
  },
  {
    "movieId": 13,
    "reviews":[{"id":1,"user":{"id":12,"name":"Isaac"},"rating":9.2,"comment":"Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.","date":"2024-01-09T07:19:48.519Z"}]
  },
  {
    "movieId": 14,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":6.8,"comment":"Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.","date":"2024-01-05T07:19:48.519Z"},{"id":2,"user":{"id":16,"name":"Mollie"},"rating":7.2,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-26T07:19:48.519Z"}]
  },
  {
    "movieId": 15,
    "reviews":[{"id":1,"user":{"id":18,"name":"Sophie"},"rating":1.1,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-21T07:19:48.519Z"}]
  },
  {
    "movieId": 16,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":2.4,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-11T07:19:48.519Z"},{"id":2,"user":{"id":14,"name":"Corey"},"rating":4.9,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-22T07:19:48.519Z"},{"id":3,"user":{"id":17,"name":"Emely"},"rating":3.4,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-30T07:19:48.519Z"}]
  },
  {
    "movieId": 17,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":5.6,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-01-10T07:19:48.519Z"},{"id":2,"user":{"id":17,"name":"Emely"},"rating":6.5,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-01-06T07:19:48.519Z"}]
  },
  {
    "movieId": 18,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":2.6,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-08T07:19:48.519Z"},{"id":2,"user":{"id":14,"name":"Corey"},"rating":4.1,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-13T07:19:48.519Z"}]
  },
  {
    "movieId": 19,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":4.9,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-01-11T07:19:48.519Z"}]
  },
  {
    "movieId": 20,
    "reviews":[{"id":1,"user":{"id":14,"name":"Corey"},"rating":8.1,"comment":"Unfortunately we don't have a reliable way to tell the true ratings of a movie.","date":"2024-02-01T07:19:48.519Z"},{"id":2,"user":{"id":11,"name":"Jack"},"rating":6.4,"comment":"It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.","date":"2024-01-21T07:19:48.519Z"}]
  },
  {
    "movieId": 21,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":6.3,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-26T07:19:48.519Z"},{"id":2,"user":{"id":12,"name":"Isaac"},"rating":1.5,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-31T07:19:48.519Z"},{"id":3,"user":{"id":10,"name":"Max"},"rating":2.6,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-30T07:19:48.519Z"}]
  },
  {
    "movieId": 22,
    "reviews":[{"id":1,"user":{"id":10,"name":"Max"},"rating":8.5,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-20T07:19:48.519Z"},{"id":2,"user":{"id":16,"name":"Mollie"},"rating":7.1,"comment":"Unfortunately we don't have a reliable way to tell the true ratings of a movie.","date":"2024-01-31T07:19:48.519Z"},{"id":3,"user":{"id":16,"name":"Mollie"},"rating":4.2,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-21T07:19:48.519Z"}]
  },
  {
    "movieId": 23,
    "reviews":[{"id":1,"user":{"id":16,"name":"Mollie"},"rating":4.5,"comment":"It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.","date":"2024-02-01T07:19:48.519Z"},{"id":2,"user":{"id":13,"name":"Zak"},"rating":1.6,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-16T07:19:48.519Z"},{"id":3,"user":{"id":14,"name":"Corey"},"rating":2,"comment":"Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.","date":"2024-01-18T07:19:48.519Z"}]
  },
  {
    "movieId": 24,
    "reviews":[{"id":1,"user":{"id":19,"name":"Christina"},"rating":9.4,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-02-01T07:19:48.519Z"}]
  },
  {
    "movieId": 25,
    "reviews":[{"id":1,"user":{"id":14,"name":"Corey"},"rating":3.7,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-18T07:19:48.519Z"},{"id":2,"user":{"id":11,"name":"Jack"},"rating":6.2,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-10T07:19:48.519Z"}]
  }
]

