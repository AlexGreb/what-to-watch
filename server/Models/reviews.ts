import { Schema, model } from 'mongoose';
import {IUser} from "./user";

type User = {
    id: number;
    name: string
}

export interface IReview {
    id: number;
    user: User;
    rating: number;
    comment: string;
}

export type MovieReviews = Array<IReview>;

export interface IMovieReviews {
    movieId: number;
    reviews: MovieReviews
}

const MovieReviewsScheme = new Schema({
    movieId: {type: Number, unique: true, required: true},
    reviews: [{
        id: {type: Number, unique: true, required: true},
        user: {
            id: {type: Number, unique: true, required: true},
            name: {type: String, unique: true, required: true}
        },
        rating: {type: String, required: true},
        comment: {type: String, required: true}
    }]
})

export const MovieReviewsModel = model<IMovieReviews>('reviews', MovieReviewsScheme);

