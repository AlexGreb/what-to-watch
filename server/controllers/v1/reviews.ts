import {Request, Response} from "express";
import type {MovieReviews} from "../../Models/reviews";
import * as reviewsService from "../../services/v1/reviewsService";
import {getResError} from "../../utils";

export const getMovieReviews = async (req: Request, res: Response): Promise<void> => {
    const {params} = req;
    if(!params.movieId) return;
    try {
        const reviews: MovieReviews | null = await reviewsService.getMovieReviews(Number(params.movieId));
        res.json({
            status: 'OK',
            data: reviews
        });
    } catch (error) {
        const {status, data} = getResError(error);
        res.status(status).json({
            status: 'FAILED',
            data
        });
    }
}
