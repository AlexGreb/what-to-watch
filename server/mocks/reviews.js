export const reviews = [
  {
    "id": 1,
    "user": {
      "id": 12,
      "name": `Isaac`
    },
    "rating": 3.6,
    "comment": `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
    "date": `2021-03-19T13:11:03.998Z`
  },
  {
    "id": 2,
    "user": {
      "id": 17,
      "name": `Emely`
    },
    "rating": 4.5,
    "comment": `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    "date": `2021-03-05T13:11:03.999Z`
  },
  {
    "id": 3,
    "user": {
      "id": 11,
      "name": `Jack`
    },
    "rating": 4.7,
    "comment": `The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. `,
    "date": `2021-03-21T13:11:03.999Z`
  }
];

export const review = [
  {
    "id": 1,
    "user": {
      "id": 12,
      "name": `Isaac`
    },
    "rating": 3.6,
    "comment": `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
    "date": `2021-03-19T13:11:03.998Z`
  },
];
