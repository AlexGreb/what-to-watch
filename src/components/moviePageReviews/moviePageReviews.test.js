/**
 * @jest-environment jsdom
 */
import React from 'react';
import {render} from '@testing-library/react';
import MoviePageReviews from './moviePageReviews.jsx';
import {review} from '../../mocks/reviews.js';

describe(`MoviePageReviews`, () => {
  it(`Should render correctly a MoviePageReviews component`, () => {
    const {asFragment} = render(<MoviePageReviews reviews={review}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

