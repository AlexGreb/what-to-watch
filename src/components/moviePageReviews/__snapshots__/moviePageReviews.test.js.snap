// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageReviews Should render correctly a MoviePageReviews component 1`] = `
<DocumentFragment>
  <div
    class="movie-card__reviews movie-card__row"
  >
    <div
      class="movie-card__reviews-col"
    >
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Unfortunately we don't have a reliable way to tell the true ratings of a movie.
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Isaac
            </cite>
            <time
              class="review__date"
              datetime="2016-12-24"
            >
              2021-03-19T13:11:03.998Z
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          3.6
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
